components/push/src/crypto.rs

```diff
@@ -50,14 +48,12 @@ impl Key {
         }
     }
 
-    pub fn key_pair(&self) -> error::Result<RcCryptoLocalKeyPair> {
-        RcCryptoLocalKeyPair::from_raw_components(&self.p256key).map_err(|e| {
```

Comment:
I think we were accidentally biting ourselves here with a bit of bad API design, which meant that we couldn't use the `aesgcm` scheme without knowing which specific crypto backend was in use by the `rust-ece` crate. That has been fixed now (thanks to an external contributor! ref https://github.com/mozilla/rust-ece/pull/47) so we can remove quite a bit of key parsing-and-re-serializing here.